### PR TITLE
Add TLS protocol parser

### PR DESCRIPTION
**Pull request by ddelnano**
Summary: Add TLS protocol parser

This is the first piece of supporting TLS protocol tracing.

Relevant Issues: N/A

Type of change: /kind feature

Test Plan: Parser test coverage verifies parser works for TLS v1.2 and later

### CODE REVIEW

File: src/stirling/source_connectors/socket_tracer/protocols/tls/parse.cc [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                 return error::Internal("Failed to extract server name"));
+
+    server_names.push_back(std::string(server_name));
+    server_name_list_length -= kSNIExtensionMinimumLength + server_name_length;
+  }
+  exts->insert({"server_name", ToJSONString(server_names)});
+  return ParseState::kSuccess;
+}
+
+ParseState ParseFullFrame(BinaryDecoder* decoder, Frame* frame) {
```
Comment (oazizi000) [inline src/stirling/source_connectors/socket_tracer/protocols/tls/parse.cc:66]: If you have it, a pointer to the protocol's wire format specification would be useful to include here.

**Review by oazizi000 — APPROVED** [review]

File: src/stirling/source_connectors/socket_tracer/protocols/tls/types.h [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+#include <vector>
+
+#include "src/common/json/json.h"
+#include "src/stirling/source_connectors/socket_tracer/protocols/common/event_parser.h"
+#include "src/stirling/utils/utils.h"
+
+namespace px {
+namespace stirling {
+namespace protocols {
+namespace tls {
```
[the remark this answers is not in the document]
Comment (ddelnano) [reply src/stirling/source_connectors/socket_tracer/protocols/tls/types.h:32]: Nice, I often forget that you can do that. Updated in a9e7bfe98.

File: src/stirling/source_connectors/socket_tracer/protocols/tls/parse_test.cc [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+  tls::Frame frame;
+  ParseState state = ParseFrame(message_type_t::kRequest, &frame_view, &frame);
+
+  ASSERT_EQ(frame.content_type, tls::ContentType::kHandshake);
+
+  ASSERT_EQ(frame.handshake_type, tls::HandshakeType::kServerHello);
+  ASSERT_EQ(state, ParseState::kSuccess);
+}
+
+namespace tls {}  // namespace tls
```
[the remark this answers is not in the document]
Comment (ddelnano) [reply src/stirling/source_connectors/socket_tracer/protocols/tls/parse_test.cc:353]: Removed in 58f492054.

File: src/stirling/source_connectors/socket_tracer/protocols/tls/parse.cc (hunk 1 above)
Comment (ddelnano) [reply src/stirling/source_connectors/socket_tracer/protocols/tls/parse.cc:66]: Nice catch. Added in 6d17ce666.

**Pull request merged**